### PR TITLE
Feature/sr 14 implement checkout page

### PR DESCRIPTION
**Pull request by GalacticOctopus**
Page contains form, map and product components. We also have added Zustand to keep global state in check.

### CODE REVIEW

**Review by masterHAWK99 — CHANGES_REQUESTED** [review]

File: src/components/booking/BookingList.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        <br />
+        <div className="product-list__total">
+          <p className="text-body-sm">Total Sum</p>
+          <h4 className="text-heading4 text-weight-600">{totalPrice}$</h4>
+        </div>
+        <button
+          className="btn btn-primary text-body-sm product-list__proceed"
+          type="submit"
+          disabled={formik.isSubmitting}
+          onClick={() => formik.handleSubmit}
```
[the remark this answers is not in the document]
Comment (GalacticOctopus) [reply src/components/booking/BookingList.tsx:39]: To my understanding `onClick={formik.handleSubmit}` secretly passes element's event for which we have no use. All that needs to happen is function executing.

File: src/scss/components/_square.scss [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  @include square;
+  transform: translate(-50%, -50%) rotate(90deg);
+}
+
+@media (max-width: 590px) {
+  .square {
+    width: 100%;
+    height: 96px;
+  }
+}
```
[the remark this answers is not in the document]
Comment (GalacticOctopus) [reply src/scss/components/_square.scss:37]: It's possible. The way it's done now allows us to configure the current dashed border in more ways than it's allowed by default.

File: src/pages/Booking.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  }, [currentUser]);
+
+  return (
+    <div className="booking">
+      <Formik
+        initialValues={initialValues}
+        validationSchema={checkoutSchema}
+        onSubmit={(values, { setSubmitting }) => {
+          values.items = items;
+          values.paymentMethod = selectedPayment;
```
[the remark this answers is not in the document]
Comment (GalacticOctopus) [reply src/pages/Booking.tsx:67]: Values is what is going to be submitted and we need to set it's values to currently selected states.

File: src/components/booking/form/PaymentMethod.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+      <ErrorMessage
+        className="error-message text-body-sm"
+        name="paymentMethod"
+        component="div"
+      />
+
+      <div className="form-group__icon-names">
+        <p className="text-caption text-weight-600">Card</p>
+        <p className="text-caption text-weight-600">PayPal</p>
```
Comment (masterHAWK99) [inline src/components/booking/form/PaymentMethod.tsx:61]: Those elements should be labels for radio fields above so users can choose payment method by clicking on name

File: src/components/map/PickLocationMap.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  },
+  {
+    value: "jerzego-7",
+    id: 3,
+    location: "Kraków, Jerzego Żuławskiego 7",
+    coordinates: [50.072545, 19.933861],
+  },
+];
+
+type TPickLocationMapProps = {
```
Comment (masterHAWK99) [inline src/components/map/PickLocationMap.tsx:35]: You don't need prefix `T` if you have suffix `Props`.

File: src/components/booking/BookingItem.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
+import { scooters } from "../../data/Scooters";
+import arrowRight from "../../assets/arrow-right.svg";
+import { useEffect, useState } from "react";
+import { Scooter } from "../../compiler/types";
+import useBasketStore from "../../store/useBasketStore";
+
+type TBookingItemProps = {
```
Comment (masterHAWK99) [inline src/components/booking/BookingItem.tsx:7]: You don't need prefix `T` if you have suffix `Props`.

**Review by masterHAWK99 — CHANGES_REQUESTED** [review]

**Pull request merged**